ratis-common/src/main/java/org/apache/ratis/util/Preconditions.java

```diff
@@ -102,6 +102,11 @@ static <T> T assertNotNull(T object, String name) {
         + name + " = " + object + " == null, class = " + object.getClass());
   }
 
+  static <T> T assertNotNull(T object, String format, Object... args) {
```

Comment:
FYI, we usually use  `Objects.requireNonNull(..)`.  It is fine if we add this method.

Reply:
Good to know. I made the decision since `Objects.requireNonNull` throws `NullPointerException` while these methods throws `IllegalStateException`. Perhaps Guava's `Precondition.checkArgument` fits better XD